src/plugins/discover/public/datasources/default_datasource.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObject } from '../../../saved_objects/public';
+import { IndexPatternSavedObjectAttrs } from '../../../data/common/index_patterns/index_patterns';
+import { IndexPatternsContract } from '../../../data/public';
+import { DataSource } from '../../../data/public';
+
+interface DataSourceConfig {
+  name: string;
+  type: string;
+  metadata: any;
+  indexPatterns: IndexPatternsContract;
+}
+
+export class DefaultDslDataSource extends DataSource<
+  any,
+  any,
+  Promise<Array<SavedObject<IndexPatternSavedObjectAttrs>> | null | undefined>,
+  any,
+  any
```

Comment:
Can you type this correctly?

Reply:
this and below related typing will be addressed after this PR merge, Tracking [issue](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5211).